dags/python_scripts/cr3_extract_diagram_ocr_narrative.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python3
+
+import io
+import os
+import re
+import sys
+import json
+import boto3
+import argparse
+import requests
+from   pdf2image import convert_from_path, convert_from_bytes
+import pytesseract
+
+# Configure the available arguments to the program
+parser = argparse.ArgumentParser(description='Extract CR3 diagrams and narratives')
+parser.add_argument('-v', action='store_true', help='Be verbose')
+parser.add_argument('-d', action='store_true', help='Check for digitally created PDFs')
+parser.add_argument('--cr3-source', metavar=('bucket', 'path'), nargs=2, required=True, help='Where can we hope to find CR3 files on S3?')
+parser.add_argument('--batch-size', metavar='int', default=1, help='How many cr3s to attempt to process?')
+parser.add_argument('--update-narrative', action='store_true', help='Update narrative in database')
+parser.add_argument('--update-timestamp', action='store_true', help='Update timestamp in database')
+parser.add_argument('--save-diagram', metavar=('bucket', 'path'), nargs=2, help='Save diagram PNG in a S3 bucket and path')
+parser.add_argument('--crash-id', metavar='int', type=int, nargs=1, default=[0], help='Specific crash ID to operate on')
+args = parser.parse_args()
+
+s3 = boto3.client('s3')
+
+def update_crash_processed_date(crash_id: int) -> bool:
+    if (args.v):
```

Comment:
Thank you for providing type annotations in all your function implementations! 📜 

I'd also appreciate if you could add docstrings to very briefly describe what each function does using this pattern:

```
:param param_type param_name: parameter description (type description is on the same line as the parameter description).
```

![](https://www.jetbrains.com/help/img/idea/2020.2/docstring_example_100.png)

Taken from: https://www.jetbrains.com/help/pycharm/using-docstrings-to-specify-types.html#example